rgen/rgen-placer/src/placer/complexplants/jungle_grass.rs

```diff
@@ -0,0 +1,307 @@
+//use core::f64::math::sqrt;
+use crate::{Placer, Random, Result, Rng};
+use rgen_base::{BlockFilter, BlockState, Pos};
+use rgen_world::{PartialWorld,};
+use std::collections::HashMap;
+//use std::collections::HashMap;
+//use std::collections::{HashSet, VecDeque}; //rng::Random
+
+// Special traits
+pub trait Style {
+    fn style(style: FloorStyle) -> Self;
+}
+
+pub enum FloorStyle {
+    Normal,
+    CanopiedJungle,
+    FlowerCanopiedJungle,
+    LightJungle,
+    TerracedJungle
+
+}
+
+pub struct JungleFloorPlace {
+  pub place_above:        BlockFilter,
+  pub place:              BlockState,
+  pub is_large:           bool,
+  pub attempts:           u32,
+  pub pink_orchid:        BlockState,
+  pub passion_flower:     BlockState,
+  pub heliconia:          BlockState,
+  pub pink_heart:         BlockState,
+  pub torch_ginger:       BlockState,
+  pub orchidaceae:        BlockState,
+  pub ipomoea:            BlockState,
+  pub bromeliads:         BlockState,
+  pub ficus_elastica:     BlockState,
+  pub yellow_jungle_rose: BlockState,
+  pub bird_of_paradise:   BlockState,
```

Comment:
you don't need these. just use the `block![]` macro when creating the `flower_array` directly.